src/averageAges.js

```diff
@@ -55,7 +73,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  // const mothers = people
+  //   .filter(human => people
+  //     .some(person => person.mother === human.name));
+
+  const children = people
+    .filter(person => (onlyWithSon)
+      ? people.some(mother => person.mother === mother.name)
+        && person.sex === 'm'
+      : people.some(mother => person.mother === mother.name));
+
+  const diffAges = children.map(child =>
+    child.born - people.find(mother => mother.name === child.mother).born);
+
+  return averageAge(diffAges);
+}
+
+function averageAge(arrayOfAge) {
```

Comment:
It is recommended to name variables without binding to it's type, for example `ages` here would be better